src/_guides/language/language-tour_migrated.md

```diff
@@ -106,14 +109,28 @@ mind:
 
 -   Everything you can place in a variable is an *object*, and every
     object is an instance of a *class*. Even numbers, functions, and
-    `null` are objects. All objects inherit from the [Object][] class.
+    `null` are objects.
+    All objects except `null` inherit from the [Object][] class.
 
 -   Although Dart is strongly typed, type annotations are optional
     because Dart can infer types. In the code above, `number`
-    is inferred to be of type `int`. When you want to explicitly say
-    that no type is expected, use the type `Object`
-    (or `Object?` if the value can be `null`) or —
-    if you want to defer type checking until runtime —
+    is inferred to be of type `int`.
+
+-   If you enable [sound null safety][ns],
+    values can’t be null unless you say they can be.
+    You can make a variable nullable by
+    putting a question mark (`?`) at the end of its type.
+    For example, a variable of type `int?` might be an integer,
+    or it might be `null`.
+
+    {{site.alert.version-note}}
+      [Null safety][ns] was introduced in Dart 2.12.
+    {{site.alert.end}}
+
+-   When you want to explicitly say
+    that no type is expected, use the type `Object?`
```

Comment:
"no type is expected" sounds sort of like it forbids everything. How about "any type is allowed"?